File: frontend/js/prediction.js
async function predictLandslide() {

    const rainfall =
        parseFloat(
            document.getElementById('rainfall').value
        );

    const slope =
        parseFloat(
            document.getElementById('slope').value
        );

    const soil =
        parseFloat(
            document.getElementById('soil').value
        );

    try {

        const response = await fetch(
            'http://127.0.0.1:5000/predict',
            {
                method: 'POST',

                headers: {
                    'Content-Type': 'application/json'
                },

                body: JSON.stringify({
                    rainfall: rainfall,
                    slope: slope,
                    soil: soil
                })
            }
        );

        const data = await response.json();


        /* Show Prediction */

        document.getElementById('result').innerHTML =
            data.prediction;


        /* Result Colors */

        if (
            data.prediction === "Landslide Risk"
        ) {

            document.getElementById('result').style.color =
                "red";

        } else {

            document.getElementById('result').style.color =
                "lightgreen";
        }


        /* Update Map */

        updateMap(
            rainfall,
            slope,
            soil,
            data.prediction
        );

    } catch (error) {

        console.log(error);

        document.getElementById('result').innerHTML =
            "Prediction Error";
    }
}

File: backend/routes/prediction.py
from flask import Blueprint, request, jsonify

from model.predict import predict_landslide

from utils.twilio_alert import send_alert

prediction_bp = Blueprint('prediction', __name__)


@prediction_bp.route('/predict', methods=['POST'])
def predict():

    data = request.json

    rainfall = float(data['rainfall'])
    slope = float(data['slope'])
    soil = float(data['soil'])

    result = predict_landslide([
        rainfall,
        slope,
        soil
    ])

    if result == "Landslide Risk":
       send_alert(
           "Warning! Landslide Risk Detected"
       )

    return jsonify({
        'prediction': result
    })